webpack/assets/javascripts/react_app/components/LoginPage/LoginPageHelpers.js

```diff
@@ -0,0 +1,86 @@
+import React, { Fragment } from 'react';
+import isEmpty from 'lodash/isEmpty';
+import { translate as __ } from '../../common/I18n';
+
+// eslint-disable-next-line consistent-return
+export const formatAlert = alerts => {
+  if (!isEmpty(alerts)) {
+    const alertType = Object.keys(alerts)[0];
+    const alertMessage = alerts[alertType];
+    return {
+      type: alertType,
+      message: alertMessage,
+      show: true,
+    };
+  }
+};
+
+export const adjustProps = ({
+  token,
+  version,
+  caption,
+  backgroundUrl,
+  logoSrc,
+  alert,
+}) => {
+  const pageAlert = formatAlert(alert);
+  const isErrorAlert = pageAlert && pageAlert.type === 'error';
+  return {
+    container: {
+      backgroundUrl,
+      alert: !isErrorAlert && pageAlert,
+    },
+    header: {
+      logoSrc,
+      caption: (
+        <Fragment>
+          <h1 id="title">{__('Welcome')}</h1>
+          {version && <p id="version">{`${__('Version')} ${version}`}</p>}
+          {caption && <p id="login_text">{caption}</p>}
+        </Fragment>
+      ),
+    },
+    card: {
+      title: __('Log into your account'),
+      form: {
+        attributes: {
+          action: '/users/login',
```

Comment:
looks like this works as a native html form, is there an option to combine `redux-form` (which we have in our project already) or alternative such as `final-form`?

Reply:
Changing the form could be done in Patternfly-react if needed, 

what are the benefits of using these form libraries in your opinion?

Reply:
ATM when there are errors, the entire page is reloaded. with redux-form we can prevent this and give a much better user experience.
this page is a full react page,  why keeping the server side rendering?

Reply:
:+1: I will add ajax requests on the follow-up PR